bin/BuildPackages.sh

```diff
@@ -186,8 +208,33 @@ do
         ;;
       esac;
       ) || build_fail
+}
+
+      (buildpackage > >(tee $LOGDIR/$dir.out) 2> >(tee $LOGDIR/$dir.err >&2) )> >(tee $LOGDIR/$dir.log ) 2>&1
       # end subshell
+      # remove superfluous log files if there was no error message
+      if [ ! -s "$LOGDIR/$dir.err" ]
+        then
+        rm -f "$LOGDIR/$dir.err"
+        rm -f "$LOGDIR/$dir.out"
+      fi
     else
       echo "$dir is not a GAP package -- no PackageInfo.g"
-    fi;
-done;
+    fi
+done < <(find -L ./ -maxdepth 1 -mindepth 1 -type d | sed 's|./||')
+
+echo "" >> $LOGDIR/$FAILPKGFILE.log
+echo ""
+echo "Output logged into pkg/$LOGDIR/$LOGFILE.log"
+echo "Packages failed to build are in pkg/$LOGDIR/$FAILPKGFILE.log"
+}
+
+# Log error to .err, output to .out, everything to .log
+( build_all_packages > >(tee $LOGDIR/$LOGFILE.out) 2> >(tee $LOGDIR/$LOGFILE.err >&2) )> >( tee $LOGDIR/$LOGFILE.log ) 2>&1
+
+# remove superfluous buildpackages log files if there was no error message
+if [ ! -s "$LOGDIR/$LOGFILE.err" ]
+  then
+  rm -f "$LOGDIR/$LOGFILE.err"
+  rm -f "$LOGDIR/$LOGFILE.out"
```

Comment:
Ahh, I missed that there are three files to which the script logs. OK, that makes sense then!